doc/models/generic.rsl

```diff
@@ -0,0 +1,12 @@
+package Generic
+
+// Diagram type to be used for PlantUML or any other image format.
+type Diagram {

```

Comment:
For me this is too flexible. It needs heuristics to detect the format. 
On the other side it is too focused, as file can also be an image that is not neccesarily  a diagram.
 
My proposal is to offer explicit types like this:

 ```
type Image {
    caption   "A caption string for the image"
        String
    file_path "The image file name path."
        String
}

type PlantumlDiagram  {
    caption    "A caption string for the diagram."
        String
    diagram    "A Plantuml diagram DSL string."
        String 
}

type PlantumlFile  {
    caption    "A caption string for the diagram."
        String
    file_path "The plantuml file name path."
        String 
}
````